Add GitHub CLI Tutorial (#40471)
firstcontributions/first-contributions: gui-tool-tutorials/github-cli-tutorial.md

[![Open Source Helpers](https://www.codetriage.com/roshanjossey/first-contributions/badges/users.svg)](https://www.codetriage.com/roshanjossey/first-contributions)

# First Contributions

| <img alt="Github CLI" src="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png" width=200> | Github CLI |
| ------------------------------------------------------------ | ---------- |

It's hard. It's always hard, when you do something for the first time. Especially when you are collaborating, making mistakes isn't a comfortable thing. But open source is all about collaboration & working together. We wanted to simplify the way new open-source contributors learn & contribute for the first time.

Reading articles & watching tutorials can help, but what comes better than actually doing the stuff without messing up anything. This project aims at providing guidance & simplifying the way rookies make their first contribution. Remember the more relaxed you are, the better you learn. If you are looking for making your first contribution just follow the simple steps below. We promise you, it will be fun.

## Github CLI

> Please note, this tutorial is for MacOS. It is similar with Windows Terminal on Windows but some things may look different.
> 
> <!--
> ************************************
> ​	*** This is commented out until      ***
> ​	*** a Windows tutorial can be created***
> ************************************
> Please note, this tutorial is for MacOS. Please refer to the [Windows Tutorial]() for Sourcetree if that is what you want to use.
> -->

Reviewer: If the installation for Windows, MacOS, and Linux below is added, then this part can be deleted
```suggestion
```

Author: Alright.

> Download [Github CLI](https://cli.github.com/), then open Terminal(or any Commandline Tools). I used the [iTerm2](https://iterm2.com/).

Reviewer: ```suggestion
To install GitHub CLI on your system, visit the [official website](https://cli.github.com/)) to download it directly or use the command-line to install the package into your system.
```

> If you want to install Github CLI via Homebrew, you can download it as shown below the command line.

Reviewer: Here is the complete installation command line script from the official docs, you can just copy the markdown below

### Homebrew (MacOS, Linux)

```bash
brew install gh
```

### MacPorts (MacOS)

```bash
sudo port install gh
```

### Conda (MacOS, Linux, Windows)

```bash
conda install gh --channel conda-forge
```

Additional Conda installation options available on the [gh-feedstock page](https://github.com/conda-forge/gh-feedstock#installing-gh).

### WinGet (Windows)

```bash
winget install gh
```

### scoop (Windows)

```bash
scoop install gh
```

### Chocolatey (Windows)

```bash
choco install gh
```

> If you want to install Github CLI via Homebrew, you can download it as shown below the command line.
> 
> ~~~
> brew install gh
> ~~~

Reviewer: Here is the complete installation command line script from the official docs, you can just copy the markdown below

```suggestion
### Homebrew (MacOS, Linux)

~~~
brew install gh
~~~

### MacPorts (MacOS)

~~~
sudo port install gh
~~~

### Conda (MacOS, Linux, Windows)

~~~
conda install gh --channel conda-forge
~~~

Additional Conda installation options available on the [gh-feedstock page](https://github.com/conda-forge/gh-feedstock#installing-gh).

### WinGet (Windows)

~~~
winget install gh
~~~

### scoop (Windows)

~~~
scoop install gh
~~~

### Chocolatey (Windows)

~~~
choco install gh
~~~
```

> 

Reviewer: ```suggestion
## After installing GitHub CLI
```

~~~

![Push changes into repository forked](https://user-images.githubusercontent.com/33862991/123590173-3fc1ff00-d825-11eb-9980-66bb309f2c89.png)

## Submit your changes for review

Finally you can submit review your commit to repository. 

~~~
gh pr create [repository-owner]/[repository-name]
~~~

`-R` or `--repo` is command for target repository.

![Source : Github CLI](https://user-images.githubusercontent.com/33862991/123590175-40f32c00-d825-11eb-92c5-af0e6ba637fd.png)

You can check PR list.

~~~
gh pr list
~~~

![PR List in console](https://user-images.githubusercontent.com/33862991/123590174-405a9580-d825-11eb-8c2a-e607d2aa192f.png)

Check your pr in web browser.

![PR List in web view](https://user-images.githubusercontent.com/33862991/123590177-40f32c00-d825-11eb-81e9-66a2a9d79e3b.png)

If your commit has not problem, It will be merging all your changes into the master branch of this project. And then you will get a notification email once the changes have been merged.

## Where to go from here?

Congrats!  You have just completed the standard _fork -> clone -> edit -> PR_ workflow that you'll encounter often as a contributor!

> Celebrate your contribution and share it with your friends and followers by going to [web app](https://firstcontributions.github.io/#social-share).

Reviewer: ```suggestion
Celebrate your contribution and share it with your friends and followers by going to [our website](https://firstcontributions.github.io/#social-share).
```

You could join our slack team in case you need any help or have any questions. [Join slack team](https://join.slack.com/t/firstcontributors/shared_invite/enQtMzE1MTYwNzI3ODQ0LTZiMDA2OGI2NTYyNjM1MTFiNTc4YTRhZTg4OWZjMzA0ZWZmY2UxYzVkMzI1ZmVmOWI4ODdkZWQwNTM2NDVmNjY).

> 

Reviewer: Add a link towards the official documentation, as follows (doesn't need to be similar but feel free to copy)

```
For more advanced command-line instructions from GitHub CLI, visit the [official documentation](https://cli.github.com/manual/).
```
